### DISCUSSION TITLE
swscale slice threading

### DISCUSSION THREAD

**Discussion by alatteri** [CESNET/UltraGrid#211] [Ideas]
Hi,

New in FFMPEG 5.0 is swscale slice threading.  Wondering if this would help performance on the color conversions that happen sometimes within UG.

Thanks,
Alan

**Comment by MartinPulec** [comment]
Hi Alan, I guess so. I've looked on it and it seems that it is not enabled by default. I think I can enable it quite easily. Anyways, I think that usually it is better to use UltraGrid own conversions so feel free to let me know if you encounter the use of swscale.

**Reply by MartinPulec** [reply]
FYI, it is already implemented [here](https://github.com/CESNET/UltraGrid/commit/cee4e8ff24662269cc96774681f2460cfe39ecba). I haven't tested it but I believe that this is correct since it is a very simple change. It should work with current AppImage and later also with other platforms' builds.

**Comment by alatteri** [comment]
Thanks Martin.  You are right, the native color conversions are probably better, but can't hurt to have this new change either.
